modules/tinymce/src/core/main/ts/fmt/FontInfo.ts

```diff
@@ -5,26 +5,54 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Fun, Obj, Optional } from '@ephox/katamari';
-import { Attribute, Compare, Css, SugarElement, SugarNode, TransformFind } from '@ephox/sugar';
+import { Arr, Obj, Optional, Fun } from '@ephox/katamari';
+import { Attribute, Css, SugarElement, SugarNode } from '@ephox/sugar';
 import DOMUtils from '../api/dom/DOMUtils';
+import * as NodeType from '../dom/NodeType';
+import * as RangeWalk from '../selection/RangeWalk';
+import { isNode } from './FormatUtils';
 
-const legacyPropNames: Record<string, string> = {
+export type FontProp = 'font-size' | 'font-family';
+
+const legacyPropNames: Record<FontProp, string> = {
   'font-size': 'size',
   'font-family': 'face'
 };
 
-const getSpecifiedFontProp = (propName: string, rootElm: Element, elm: HTMLElement): Optional<string> => {
-  const getProperty = (elm: SugarElement) => Css.getRaw(elm, propName).orThunk(() => {
-    if (SugarNode.name(elm) === 'font') {
-      return Obj.get(legacyPropNames, propName).bind((legacyPropName) => Attribute.getOpt(elm, legacyPropName));
-    } else {
-      return Optional.none();
-    }
-  });
-  const isRoot = (elm: SugarElement) => Compare.eq(SugarElement.fromDom(rootElm), elm);
+const collect = (dom: DOMUtils, rngOrNode: Range | Node, rootNode: Element, propName: FontProp): string[] => {
+  const collector: string[] = [];
+
+  const process = (nodes: Node[], limit: Node, getComputed: boolean) => {
```

Comment:
`limit` is a rather strange name for a node, I'd suggest calling this `root` instead, as that's what we use in almost every other place.

Reply:
Also from what I can see, you don't need a `Node[]` here and instead just require an `ArrayLike<Node>` array instead. That'll save having to do the `Arr.from(node.childNodes)` below as well. I could have very well missed something though :shrug: